### PR TITLE
Reconfigured the security action tips

### PR DESCRIPTION
**Pull request by houghj16**
### Motivation
The existing security action tips were outdated and no longer aligned with our current security practices. This pull request aims to enhance the security posture of our application by reconfiguring the action tips to reflect the latest best practices.

### Description of Changes
- Updated Configuration Files: Modified the configuration files related to security action tips.
- Revised Content: Rewrote the content of the security action tips to provide clearer instructions and emphasize critical security measures.
- Added Examples: Included practical examples to illustrate each security tip.
- Removed Deprecated Tips: Removed any deprecated or redundant tips.

 ### Testing Information
- Unit Tests: Added new unit tests to verify the correctness of the updated security action tips.
- Integration Testing: Conducted integration testing to ensure that the reconfigured tips integrate seamlessly with the existing security features.
- Manual Testing: Manually tested the application with the revised action tips to validate their effectiveness.

This pull request primarily includes changes related to the renaming of a class, the addition of a new property to a class, and the addition of new settings. The class `QRCodeForm` has been renamed to `QRCodeFormItem` in multiple files and all its instances have been updated accordingly. A new property `Image` has been added to the `BalloonTipAction` class. Additionally, a new setting has been added to the `.vscode/settings.json` file and a new `SECURITY.md` file has been created.

Here are the most important changes:

Renaming of `QRCodeForm` to `QRCodeFormItem`:

* [`ShareX/Forms/QRCodeForm.Designer.cs`](diffhunk://#diff-51fd53be88b0e28f14b1753a347a7037db0c394e31bd995ac1041d8e3909214cL3-R3): The class `QRCodeForm` was renamed to `QRCodeFormItem` and all instances of `QRCodeForm` were replaced with `QRCodeFormItem`. This change was also reflected in the `InitializeComponent` method where the `resources` object now uses `QRCodeFormItem` instead of `QRCodeForm`. [[1]](diffhunk://#diff-51fd53be88b0e28f14b1753a347a7037db0c394e31bd995ac1041d8e3909214cL3-R3) [[2]](diffhunk://#diff-51fd53be88b0e28f14b1753a347a7037db0c394e31bd995ac1041d8e3909214cL31-R31)
* [`ShareX/Forms/QRCodeForm.cs`](diffhunk://#diff-130279d4888be04d3e9b355a9aed3cf1d01e1d47aa49333e660a70a8b263b9d7L41-R55): The class `QRCodeForm` was renamed to `QRCodeFormItem` and all instances of `QRCodeForm` were replaced with `QRCodeFormItem`. This includes changes in the constructor, `Instance` property, and several static methods. [[1]](diffhunk://#diff-130279d4888be04d3e9b355a9aed3cf1d01e1d47aa49333e660a70a8b263b9d7L41-R55) [[2]](diffhunk://#diff-130279d4888be04d3e9b355a9aed3cf1d01e1d47aa49333e660a70a8b263b9d7L60-R64) [[3]](diffhunk://#diff-130279d4888be04d3e9b355a9aed3cf1d01e1d47aa49333e660a70a8b263b9d7L71-R97)
* [`ShareX/TaskHelpers.cs`](diffhunk://#diff-47e32724649d27c7e068c9fd6eea299acb6b59ba5a0510447b463fa629e2283dL1261-R1266): The static methods `OpenQRCode` and `OpenQRCodeDecodeFromScreen` were updated to use `QRCodeFormItem` instead of `QRCodeForm`.
* [`ShareX/UploadInfoManager.cs`](diffhunk://#diff-eb8a054d84a6eb8ac7979f675c5cc7dfb29d30c10d357f0e215260f01bb96805L393-R393): The `ShowQRCode` method was updated to use `QRCodeFormItem` instead of `QRCodeForm`.
* [`ShareX/WorkerTask.cs`](diffhunk://#diff-c103f6f8808da9f1966e1c6efad17382e0234b0216d56bfd888f2e3470dcf385L815-R815): The `DoFileJobs` and `DoAfterUploadJobs` methods were updated to use `QRCodeFormItem` instead of `QRCodeForm`. [[1]](diffhunk://#diff-c103f6f8808da9f1966e1c6efad17382e0234b0216d56bfd888f2e3470dcf385L815-R815) [[2]](diffhunk://#diff-c103f6f8808da9f1966e1c6efad17382e0234b0216d56bfd888f2e3470dcf385L919-R919)

Addition of new property:

* [`ShareX/BalloonTipAction.cs`](diffhunk://#diff-0c1885a7bf9c738b354d8eb7bc9773374fc0f4525c73a3190cadc3e069c32d64R32-R33): Added a new property `Image` to the `BalloonTipAction` class.

Minor changes:

* [`.vscode/settings.json`](diffhunk://#diff-a5de3e5871ffcc383a2294845bd3df25d3eeff6c29ad46e3a396577c413bf357R1-R3): Added a new setting `dotnet.preferCSharpExtension` and set it to `true`.
* [`SECURITY.md`](diffhunk://#diff-f6ed156e4bf5c791680662464b94ea5d753f219ee816b385f67870e2c0d7d4c7R1-R16): Added a new file with details on how to report a security vulnerability.
* [`ShareX/EasterEggBounce.cs`](diffhunk://#diff-38ac520f3b69312ecc4b764b32c15e2274287be9b15588c7c056d10c29b2ff23R32-R37): Added some empty lines to the file.
* [`ShareX/TaskManager.cs`](diffhunk://#diff-55a3db35892296f165ba0f17e4d9250bcfde544a852eb97901d8078f8e532fb3L53-R53): Added a comment on the line where a task is added to the `Tasks` list.

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Security Policy
+
+## Reporting a Vulnerability
+
+We take the security of our project seriously. If you discover a security vulnerability, we appreciate your help in disclosing it to us in a responsible manner.
+
+To report a security vulnerability, please follow these steps:
+
+1. Visit our [Security Page](https://github.com/ShareX/ShareX/security) on GitHub.
+2. Click on the "Report a vulnerability" button.
```
Comment (houghj16) [inline SECURITY.md:10]: Comment for review

File: SECURITY.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+3. Provide a detailed description of the vulnerability, including steps to reproduce if possible.
+4. Allow us a reasonable amount of time to address and mitigate the issue before disclosing it publicly.
+
+We encourage you to play a crucial role in keeping our project secure, and we thank you for your contributions to the community.
```
Comment (houghj16) [inline SECURITY.md:14]: Comment 2 for review

**Review by houghj16 — COMMENTED** [review]
Please disregard items 2 and 3 on the attached ticket, we'll be addressing those in a separate PR.

File: ShareX/BalloonTipAction.cs [hunk 3]
```diff
@@ -29,5 +29,7 @@ public class BalloonTipAction
     {
         public BalloonTipClickAction ClickAction { get; set; }
         public string Text { get; set; }
+
```
Comment (otheruser16) [inline ShareX/BalloonTipAction.cs:32]: Comment for review

**Review by otheruser16 — APPROVED** [review]

File: ShareX/EasterEggBounce.cs [hunk 4]
```diff
@@ -29,6 +29,12 @@
 using System.Windows.Forms;
 
 namespace ShareX
+
+
+
+
```
Comment (otheruser16) [inline ShareX/EasterEggBounce.cs:35]: Comment for review

**Review by otheruser16 — CHANGES_REQUESTED** [review]
Request changes

File: SECURITY.md (hunk 2 above)
Comment (houghj16) [reply SECURITY.md:14]: Thanks!

File: ShareX/BalloonTipAction.cs (hunk 3 above)
Comment (houghj16) [inline ShareX/BalloonTipAction.cs:32]: Add image Object

File: SECURITY.md (hunk 2 above)
Comment (houghj16) [reply SECURITY.md:14]: Thanks!

File: ShareX/Forms/QRCodeForm.cs [hunk 5]
```diff
@@ -57,7 +61,7 @@ public static QRCodeForm Instance
 
         private bool isReady;
 
-        public QRCodeForm(string text = null)
+        public QRCodeFormItem(string text = null)
```
Comment (houghj16) [inline ShareX/Forms/QRCodeForm.cs:64]: Use a more descriptive label for the text

File: ShareX/Forms/QRCodeForm.Designer.cs [hunk 6]
```diff
@@ -1,6 +1,6 @@
 ﻿namespace ShareX
 {
-    partial class QRCodeForm
+    partial class QRCodeFormItem
     {
         /// <summary>
```
Comment (houghj16) [inline ShareX/Forms/QRCodeForm.Designer.cs:5]: Can we add more explanation here?

File: ShareX/BalloonTipAction.cs [hunk 7]
```diff
@@ -29,5 +29,7 @@ public class BalloonTipAction
     {
         public BalloonTipClickAction ClickAction { get; set; }
         public string Text { get; set; }
+        public object Image { get; set; }
```
Comment (otheruser16) [inline ShareX/BalloonTipAction.cs:32]: can we use a different type for this image variable

**Pull request closed without merging**